x-pack/plugins/enterprise_search/public/applications/analytics/components/layout/page_template.tsx

```diff
@@ -27,26 +33,60 @@ interface EnterpriseSearchAnalyticsPageTemplateProps extends PageTemplateProps {
 export const EnterpriseSearchAnalyticsPageTemplate: React.FC<
   EnterpriseSearchAnalyticsPageTemplateProps
 > = ({ children, analyticsName, pageChrome, pageViewTelemetry, ...pageTemplateProps }) => {
+  const { updateSideNavDefinition, getChromeStyle$ } = useValues(KibanaLogic);
+  const chromeStyle = useObservable(getChromeStyle$(), 'classic');
+  const alwaysReturnNavItems = true;
+
+  const navItems = useEnterpriseSearchAnalyticsNav(
+    analyticsName,
+    analyticsName
+      ? {
+          explorer: generateEncodedPath(COLLECTION_EXPLORER_PATH, {
+            name: analyticsName,
+          }),
+          integration: generateEncodedPath(COLLECTION_INTEGRATE_PATH, {
+            name: analyticsName,
+          }),
+          overview: generateEncodedPath(COLLECTION_OVERVIEW_PATH, {
+            name: analyticsName,
+          }),
+        }
+      : undefined,
+    alwaysReturnNavItems
+  );
+
+  const getSelectedCollectionItems = useCallback(
+    (
+      items?: Array<EuiSideNavItemTypeEnhanced<unknown>>
+    ): Array<EuiSideNavItemTypeEnhanced<unknown>> | undefined => {
+      if (!items) return undefined;
+
+      const buildGroup = items.find((item) => item.id === 'build');
+      if (!buildGroup || !buildGroup.items) return undefined;
+
+      const searchAppsGroup = buildGroup.items.find((item) => item.id === 'analyticsCollections');
+
+      return searchAppsGroup?.items;
+    },
+    []
+  );
+
+  useEffect(() => {
+    // We update the new side nav definition with the selected collection items
+    updateSideNavDefinition({ collections: getSelectedCollectionItems(navItems) });
+  }, [navItems, updateSideNavDefinition]);
+
+  useEffect(() => {
+    return () => {
+      updateSideNavDefinition({ collections: undefined });
```

Comment:
I'm not sure why it is in a separate useEffect and not cleaned up on the previous one. 

maybe the intention was to clean up only on unmount? but then the dependencies need to be `[updateSideNavDefinition]`

Reply:
I put it in a separate useEffect as I only want `updateSideNavDefinition` to be called with `undefined` when the comp unmounts. Otherwise we would get 2 calls each time `navItems` changes, right? Or does React handles this case and merges the 2 calls into 1?

Great catch for the useEffect dependency, I removed it in https://github.com/elastic/kibana/pull/183157/commits/3271e19ea024d76b773b1a25139405d5c96867b6

Reply:
React should handle this and only call this function a single time when the component is unmounted.

Reply:
Apparently it does have a slight effect and calls the cleaning function on more time

### Everytime `navItems` changes

<img width="177" alt="Screenshot 2024-05-22 at 16 52 05" src="https://github.com/elastic/kibana/assets/2854616/7ba9f0df-b1b7-434f-bdb2-7071645932cc">


### Only when component unmounts
<img width="177" alt="Screenshot 2024-05-22 at 16 52 40" src="https://github.com/elastic/kibana/assets/2854616/61adc4c4-b5d5-4488-8108-22decc731988">

I am going to leave it as is for now if that's OK.